src/veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h

```diff
@@ -0,0 +1,153 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#ifndef SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_PERSONSUBSCRIPTIONMANAGER_H_
+#define SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_PERSONSUBSCRIPTIONMANAGER_H_
+
+#include <set>
+#include <string>
+#include <list>
+
+#include "veins/modules/mobility/traci/subscriptionManagement/SubscriptionManagerBase.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIPerson.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+/**
+ * @class PersonSubscriptionManager
+ *
+ * Manages all person subscriptions. You can update it via a subscription
+ * result in the form of a TraCIBuffer or via a manual update but it is
+ * necessary to provide an id_list of active persons then.
+ *
+ * After updating you can request the disappeared and appeared persons via
+ * the class methods.
+ *
+ * IMPORTANT: Call initialize() before using!
+ *
+ * @author Nico Dassler <dassler@hm.edu>
+ */
+class PersonSubscriptionManager: public SubscriptionManagerBase {
+
+public:
+    /**
+     * Constructor.
+     */
+    PersonSubscriptionManager();
+
+    /**
+     * Default destructor.
+     */
+    virtual ~PersonSubscriptionManager() = default;
+
+    /**
+     * Update this PersonSubscriptionManager with the given
+     * active person id list. You can force an update of active
+     * persons using this.
+     *
+     * @param currentlyActivePersonIds A list containing the currently active persons
+     * identified by their ids.
+     *
+     * @return true
+     */
+    bool updateWithList(std::list<std::string>& currentlyActivePersonIds);
+
+    /**
+     * Update this manager with the given buffer containing a subscription
+     * response for a person value.
+     *
+     * This will create new subscriptions for specific persons if the buffer
+     * contained an ID_LIST subscription.
+     *
+     * @param buffer the traci buffer containing the subscription response.
+     *
+     * @return bool true if the given buffer contained an ID_LIST response.
+     */
+    bool update(TraCIBuffer& buffer) override;
+
+    /**
+     * This gives you all the persons that were updated (includes
+     * new persons) since the last time you called this method. That
+     * means that multiple calls to this method will NOT yield the same
+     * result.
+     *
+     * This can also contain multiple updates for a single person.
+     *
+     * @return a list of TraCIPerson.
+     */
+    std::list<TraCIPerson> getUpdated();
```

Comment:
Any particular (and good) reason to use a `std::list` here? If not,just us a `std::vector`.

Reply:
modified.